### PR TITLE
media-sound/lilypond-2.18.2-r2: correct slot dep on guile and resolve #586592

### PR DESCRIPTION
**Pull request by lilrc**
This is a bugfix commit for lilypond-2.18.2-r2, which I want to get stabilized. This corrects the slot dependency on guile, so that the correct version is pulled in (>=dev-scheme/guile-2.0.0 does not work). Also Gentoo bug #586592 is fixed by calling xdg_environment_reset in src_prepare.

### CODE REVIEW

**Comment by mgorny** [comment]
@gentoo/scheme, is it a good idea to dep on subslot like that?

**Comment by radhermit** [comment]
No, just version restrict it.

File: media-sound/lilypond/lilypond-2.18.2-r2.ebuild [hunk 1]
```diff
@@ -19,7 +19,7 @@ IUSE="debug emacs profile vim-syntax ${LANGS// / linguas_}"
 REQUIRED_USE="${PYTHON_REQUIRED_USE}"
 
 RDEPEND=">=app-text/ghostscript-gpl-8.15
-	>=dev-scheme/guile-1.8.2:12[deprecated,regex]
+	>=dev-scheme/guile-1.8.2:12/8[deprecated,regex]
```
Comment (mgorny) [inline media-sound/lilypond/lilypond-2.18.2-r2.ebuild]: What he said. Just add `<dev-scheme/guile-2.0:12` instead.

**Comment by lilrc** [comment]
@mgorny

**Comment by mgorny** [comment]
Could you rebase, please? It's been bumped 2 days ago already ;-).

**Comment by lilrc** [comment]
@mgorny Rebase done.

**Comment by mgorny** [comment]
@lilrc, ok, I've looked at the detailed diff, and another question: any reason to force a USE flag instead of just using whichever guile is installed in the newer versions? You could use `guile:=` to force rebuilding lilypond when switching from guile-1 to -2.

**Comment by lilrc** [comment]
@mgorny The USE flag makes it explicit that guile-2 is experimental. Furthermore, the user who just runs `emerge lilypond` on ~arch will not get the experimental software by default. Instead the ebuild will pull in a more sensible default guile version. See for example Gentoo bug #538592.

**Comment by mgorny** [comment]
Ok. I can't say that I like it but it's god enough for me.

**Pull request closed without merging**